packages/bsky/tests/views/author-feed.test.ts

```diff
@@ -386,6 +388,142 @@ describe('pds author feed views', () => {
+        {
+          headers: sc.getHeaders(alice),
+        },
+      )
+      await network.processAll()
+
+      return post
+    }
+
+    it('params.includePins = true, pin is in first page of results', async () => {
+      const post = await createAndPinPost()
+      const { data } = await agent.api.app.bsky.feed.getAuthorFeed(
+        { actor: sc.accounts[alice].handle, includePins: true },
+        {
+          headers: await network.serviceHeaders(
+            alice,
+            ids.AppBskyFeedGetAuthorFeed,
+          ),
+        },
+      )
+
+      const pinnedPosts = data.feed.filter(
+        (item) => item.post.uri === post.ref.uriStr,
+      )
+      expect(pinnedPosts.length).toEqual(1)
+
+      const pinnedPost = data.feed.at(0)
+      expect(pinnedPost?.post?.uri).toEqual(post.ref.uriStr)
+      expect(pinnedPost?.post?.viewer?.pinned).toBeTruthy()
+      expect(AppBskyFeedDefs.isReasonPin(pinnedPost?.reason)).toBeTruthy()
+
+      const notPinnedPost = data.feed.at(1)
+      expect(notPinnedPost?.post?.viewer?.pinned).toBeFalsy()
+      expect(forSnapshot(data.feed)).toMatchSnapshot()
+    })
+
+    it('params.includePins = true, pin is NOT in first page of results', async () => {
+      const post = await createAndPinPost()
+      await sc.post(alice, 'not pinned post')
+      await sc.post(alice, 'not pinned post')
+      await network.processAll()
+      const { data: page1 } = await agent.api.app.bsky.feed.getAuthorFeed(
+        { actor: sc.accounts[alice].handle, includePins: true, limit: 2 },
+        {
+          headers: await network.serviceHeaders(
+            alice,
+            ids.AppBskyFeedGetAuthorFeed,
+          ),
+        },
+      )
+
+      // exists with `reason`
+      const pinnedPost = page1.feed.find(
+        (item) => item.post.uri === post.ref.uriStr,
+      )
+      expect(pinnedPost?.post?.uri).toEqual(post.ref.uriStr)
+      expect(pinnedPost?.post?.viewer?.pinned).toBeTruthy()
+      expect(AppBskyFeedDefs.isReasonPin(pinnedPost?.reason)).toBeTruthy()
+      expect(forSnapshot(page1.feed)).toMatchSnapshot()
+
+      const { data: page2 } = await agent.api.app.bsky.feed.getAuthorFeed(
+        {
+          actor: sc.accounts[alice].handle,
+          includePins: true,
+          cursor: page1.cursor,
+        },
+        {
+          headers: await network.serviceHeaders(
+            alice,
+            ids.AppBskyFeedGetAuthorFeed,
+          ),
+        },
+      )
+
+      // exists without `reason`
+      const laterPinnedPost = page2.feed.find(
+        (item) => item.post.uri === post.ref.uriStr,
+      )
+      expect(laterPinnedPost?.post?.uri).toEqual(post.ref.uriStr)
+      expect(laterPinnedPost?.post?.viewer?.pinned).toBeTruthy()
+      expect(AppBskyFeedDefs.isReasonPin(laterPinnedPost?.reason)).toBeFalsy()
+      expect(forSnapshot(page2.feed)).toMatchSnapshot()
+    })
+
+    it('params.includePins = false', async () => {
+      const post = await createAndPinPost()
```

Comment:
would be nice to create one other post here as well just to ensure that it doesn't still stick it at the top of the page